Added the possibility to translate menus and datafield labels to any language. Added Spanish Language. And other little changes.

alsama opened the pull request:
Hello:
I use this great app and want to propouse some changes in order to make app more accesible by adding other languages to menus and datafields labels. And also other little changes.
CHANGES:
Added function: "get_va_device" to determine vivoactive devices so can use the "resources-LANG" strings XML files to transalate texts.
Added the posibility to translate app texts to any language. Using: WatchUi.loadResource(Rez.Strings.XXXX) and strings.XML files.
Added the translate Language: Spanish. FILE: "resources-spa/strings/strings.xml"
Added a dialog to confirm exit app.
Added a dialog to confirm discard an activity.
Added one decimal to battery value and "%" symbol.

alsama committed:
Update README.md

CHANGES IN "alsama/wormnav" FORK

alsama committed:
Update README.md

Commit:
alsama_wormnav_v1

alsama commented on README.md:
```diff
@@ -26,4 +26,12 @@ It consists of two apps:
\ No newline at end of file
+- Potential issue with Garmin's Android BLE library on Android 11 devices
+
+**alsama/wormnav CHANGES:**
```
This lines in Readme.md could be changed to simply tell that now app is multi language or the languages that is translated.

alsama commented on connectiq/.project:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <projectDescription>
-	<name>wormnav</name>
+	<name>wormnav_almaca_v1</name>
```
This tag must be: "wormnav".

alsama commented on connectiq/source/TrackModel.mc:
```diff
@@ -10,10 +10,69 @@ class TrackModel {
 
     hidden var data;
     hidden var boundingBox;
+    
```
This can be deleted. Is just a comment for try in garmin simulator.

alsama commented on connectiq/source/TrackModel.mc:
```diff
@@ -10,10 +10,69 @@ class TrackModel {
     function initialize(msg) {
+        
         data = msg;
+        /*
```
This can be also deleted. Is just for test.

alsama commented on connectiq/source/WormNavDelegate.mc:
```diff
@@ -12,21 +12,6 @@ var mode;
 
 var dataPage = 0;
 
-class ExitConfirmationDelegate extends WatchUi.ConfirmationDelegate {
```
moved to "MainMenu.mc"

andan67 commented:
Hello,
Thank for your pull request which I really appreciate. My only concern is the increased memory usage when running on older low spec devices likey my FR235, where only 64kB (!) memory capacity is availlable for ConnectIQ apps. I will check if and how exclude annotations can be used to easily build device dependent release variants.

alsama commented:
I aggree with you any line you add to the code will increase memory usage and could make that can not load a large Route in the program. 
In my test for fr235 memory increases from 46kB to 49,6kB.
But the good is that next language translates wil not increase memory usage only device storage for language files.
You must think if to enable to translate app texts, is a thing that will be more usefull to users than that lack of memory.

But It will be great have a device dependent versions so then could add new features in devices with more memory.

Because, I am also testing some additonal code for CIQ 3.x devices like Menu2 and MenuItem Label and Sublabel Values. That is better because in sublabels it can show you selected options with out enter in each option.
Also I want to try to have several routes on device using several "trackData_#" AppProperties using device storage. And to add new menus to save and load routes...
Maybe could add too, new menus to custom some things like color or line width of route, or the gps arrow color,  etc.
or add optional help icons to the map screen for zoom in and out, etc.

When I try any that things I will send it to see if could be added in a version for devices with more memory. Or in old devices if you think that is usefull.

alsama committed:
Adjusts to Exit and discard dialog

Using separated dialog classes to EXIT and DISCARD.
Added French and German file languages using English file language just to test memory usage.

alsama reviewed (commented):
I have detected a problem when go to exit dialog from discard option and use the touch screen to exit from that dialog.
Making that steps in my vivoactive 3 (firmware 7.70), wormnav app crash and I have see that all CIQs apps installed are afected and watch must be restared to work fine again.
So I have put seperate user dialogs for EXIT and DISCARD and see that problem is solved.

Also I have add french a german languages copied from english to see that memory usage is not increased when new language files are added.

alsama committed:
function "get_va_device" modified

I have modified "get_va_device" function to evite use partNumber as it not detect some devices, and also less code lines are needed to detect vivoactive devices.
Also added French and Deutch Languages in mainfest and translated some texts wating that some french o german user will make a better traduction.

andan67 commented:
I've merged the pull request into a separate branch for testing. 

I also made a change in the handling of string arrays to reduce memory consumption (important for FR235). These can now be defined by a single string where the array items are separated by the '|' character. The array will be construvted by a split method

Example:
`["Timer", "Distance"] = Application.getApp().split("Timer|Distance")`

alsama commented:
ok.
Is a great idea process texts in that way to save memory. Make the text tranlations using that strings is also possible.
Also the new "GenericPickerDelegate" class is a good modification to save memory.
I have noticed that in new versions now is posible load biggers tracks than in older versions.
And the "Improved performance" of the last commit is a good notice but in my vivoactive 3 also was working fine before.
New Play and stop symbols are also a good for the user experience.
Congrats.

alsama commented:
Is great to see that you are working over a multi language version.
Only remember update this files:
Readme.md (I write "alsama/wormnav CHANGES:" comments just to tell you the changes. Now can be modified or deleted)
.project "projectDescription" "name" tag was changed in my test using eclipse.
